plaso/parsers/text_plugins/teamviewer.py

```diff
@@ -0,0 +1,553 @@
+from plaso.parsers.text_plugins import interface
+
+
+class TeamViewerConnectionsIncomingEventData(events.EventData):
+  """TeamViewer incoming connection log event data.
+
+  Attributes:
+    activity_type (str): Type of the activity
+        (e.g. RemoteSupport, FileTransfer).
+    connection_guid (str): The GUID associated with the connection.
+    display_name (string): The display name of the incoming connection
+        source. Usually the computer name or the TeamViewer user name.
+    local_account (str): The local user account associated
+        with this activity.
+    session_end_timestamp (dfdatetime.DateTimeValues): End time
+        of the connection (UTC).
+    session_start_timestamp (dfdatetime.DateTimeValues): Start time
+        of the connection (UTC).
+    source_teamviewer_id (int): The TeamViewer ID of
+        the incoming connection.
+  """
+
+  DATA_TYPE = 'teamviewer:connections_incoming:entry'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(TeamViewerConnectionsIncomingEventData, self).__init__(
+        data_type=self.DATA_TYPE)
+    self.activity_type = None
+    self.connection_guid = None
+    self.display_name = None
+    self.local_account = None
+    self.session_end_timestamp = None
+    self.session_start_timestamp = None
+    self.source_teamviewer_id = None
+
+
+class TeamViewerConnectionsOutgoingEventData(events.EventData):
+  """TeamViewer outgoing connection log event data.
+
+  Attributes:
+    activity_type (str): Type of the activity
+        (e.g. RemoteSupport, FileTransfer).
+    connection_guid (str): The GUID associated with the connection.
+    destination_teamviewer_id (int): The TeamViewer ID of the destination.
+    local_account (str): The local user account associated
+        with this activity.
+    session_end_timestamp (dfdatetime.DateTimeValues): End time
+        of the connection (UTC).
+    session_start_timestamp (dfdatetime.DateTimeValues): Start time
+        of the connection (UTC).
+  """
+
+  DATA_TYPE = 'teamviewer:connections_outgoing:entry'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(TeamViewerConnectionsOutgoingEventData, self).__init__(
+        data_type=self.DATA_TYPE)
+    self.activity_type = None
+    self.connection_guid = None
+    self.destination_teamviewer_id = None
+    self.local_account = None
+    self.session_end_timestamp = None
+    self.session_start_timestamp = None
+
+
+class TeamViewerApplicationEventData(events.EventData):
+  """TeamViewer application log event data.
+
+  Attributes:
+      message (str): the log message.
+      process_id (int): the process ID that generated the log entry.
+      timestamp (dfdatetime.DateTimeValues): the event timestamp.
+  """
+
+  DATA_TYPE = 'teamviewer:application_log:entry'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(TeamViewerApplicationEventData, self).__init__(
+        data_type=self.DATA_TYPE)
+    self.message = None
```

Comment:
message is a reserved output field name https://plaso.readthedocs.io/en/latest/sources/user/Output-and-formatting.html